### ISSUE TITLE
Extractions still being linted in unnecessary_lambda_linter() with explicit return

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2258]
```r
lint("
  sapply(
    x,
    function(xi) {
      return(foo(xi)$bar)
    }
  )
", unnecessary_lambda_linter())
# <text>:4:5: warning: [unnecessary_lambda_linter] Pass foo directly as a symbol to sapply() instead of wrapping it in an unnecessary anonymous function. For example, prefer lapply(DF, sum) to lapply(DF, function(x) sum(x)).
#     function(xi) {
#     ^~~~~~~~~~~~~~```

**Comment by MichaelChirico** [comment]
Another test case:

```r
apply(X, 1, function(x) {
  return(sum(x) == 0)
})
```

**Comment by MichaelChirico** [comment]
And another:

```r
sapply(
  colnames(data.wide),
  function(nm) {
    return(data[match(nm, data[, by.var]), color.var])
  }
)
```
